### PR TITLE
e2e gating

### PR DESCRIPTION
**Pull request by paulfantom**
Reuse test suite from mao to provide e2e gating tests for AWS provider.

### CODE REVIEW

**Comment by ingvagabund** [comment]
Consider moving the testsuite under actuator-pkg repo and vendoring it here. @enxebre we should move the same code from mao to actuator-pkg.

**Comment by ingvagabund** [comment]
/approve
de-duplication can be done in parallel.

File: Gopkg.toml [hunk 1]
```diff
@@ -41,3 +41,7 @@ required = [
 name = "gopkg.in/fsnotify.v1"
```
Comment (enxebre) [inline Gopkg.toml]: could you split the vendoring commits from the actual code commit?

Comment (paulfantom) [reply Gopkg.toml]: Done

File: test/e2e/main.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"flag"
+	"github.com/golang/glog"
+	osconfigv1 "github.com/openshift/api/config/v1"
```
Comment (enxebre) [inline test/e2e/main.go]: this is to test the operator status which I can't see  here, hence don't need to vendor. So either bring the test for it here or remove revendoring

File: test/e2e/main.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+const (
+	namespace = "openshift-cluster-api"
+)
+
+func init() {
+	if err := capiv1alpha1.AddToScheme(scheme.Scheme); err != nil {
+		glog.Fatal(err)
+	}
+
+	if err := osconfigv1.AddToScheme(scheme.Scheme); err != nil {
```
Comment (enxebre) [inline test/e2e/main.go]: you are adding osconfigv1 to the scheme but I can't see where it's used, see comment above

File: test/e2e/main.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	glog.Info("PASS: ExpectProviderAvailable")
+
+	glog.Info("RUN: ExpectOneClusterObject")
+	if err := testConfig.ExpectOneClusterObject(); err != nil {
+		glog.Errorf("FAIL: ExpectOneClusterObject: %v", err)
+		return err
+	}
+	glog.Info("PASS: ExpectOneClusterObject")
+
+	glog.Info("RUN: ExpectReconcileControllersDeployment")
```
Comment (enxebre) [inline test/e2e/main.go]: It's fine to have this here but this is actually validating machine api operator behaviour. We'll want to battle prove the machineSet/machine(actuator) controllers here e.g https://github.com/openshift/machine-api-operator/blob/master/test/e2e/operator_expectations.go#L89
https://github.com/openshift/machine-api-operator/blob/master/test/e2e/operator_expectations.go#L178
And also will want to include coverage for the validations we do against a vanilla cluster https://github.com/openshift/cluster-api-provider-aws/blob/master/test/machines/machines_test.go#L129

File: test/e2e/main.go (hunk 2 above)
Comment (paulfantom) [reply test/e2e/main.go]: removed

File: test/e2e/main.go (hunk 3 above)
Comment (paulfantom) [reply test/e2e/main.go]: removed

File: Makefile [hunk 5]
```diff
@@ -89,6 +89,10 @@ test-e2e: ## Run e2e test
 	# KUBECONFIG and SSH_PK dirs needs to be mounted inside a container if tests are run in containers
 	go test -timeout 20m -v sigs.k8s.io/cluster-api-provider-aws/test/machines -kubeconfig $${KUBECONFIG:-~/.kube/config} -ssh-key $${SSH_PK:-~/.ssh/id_rsa} -actuator-image $${ACTUATOR_IMAGE:-gcr.io/k8s-cluster-api/aws-machine-controller:0.0.1} -cluster-id $${ENVIRONMENT_ID:-""} -ginkgo.v
 
+.PHONY: validate-e2e
+validate-e2e: ## Run e2e validation/gating test
```
Comment (ingvagabund) [inline Makefile]: Can you extend the `make check` to run it as well?

Comment (paulfantom) [reply Makefile]: That doesn't seem right. Currently you can run `make check` without any external dependencies but `validate-e2e` needs a running cluster to be executed.

Comment (ingvagabund) [reply Makefile]: Oh, make sense. Ignore the comment :)

Comment (enxebre) [reply Makefile]: Can we don't introduce skew unless there's reason? https://github.com/openshift/machine-api-operator/blob/master/Makefile#L48

**Comment by enxebre** [comment]
lgtm other than the comment regarding the command name. Can you create the counterpart PR for the release Job with optional run so we can merge this on "Green"?

File: Makefile (hunk 5 above)
Comment (paulfantom) [reply Makefile]: `test-e2e` is already taken

Comment (enxebre) [reply Makefile]: I see so can we please (may be in different PR) make sure we rename the vanilla tests to test-*
So we use test-e2e consistently across all repos for the product e2e testing

Comment (paulfantom) [reply Makefile]: Yup, I'll do it.

Comment (paulfantom) [reply Makefile]: Done in https://github.com/openshift/cluster-api-provider-aws/pull/117

**Comment by paulfantom** [comment]
CI pipeline implementation in https://github.com/openshift/release/pull/2256

**Pull request merged**